### DISCUSSION TITLE
Lodash Debounce not working in storybook/react 6.5.9

### DISCUSSION THREAD

**Discussion by FocusCookie** [storybookjs/storybook#24760] [Help]
Hello 👋,

I'm facing the problem that the lodash debounce function is not working as expected in Storybook. I debounced an input field text. In my app, it works as expected, but not in the SB story. Also, when I set the debounce time to 0, it works in Storybook as well. The regular usage of setTimeout is working in Storybook. 

Lodash version: 4.17.15

Does anybody have an idea what could cause this behavior? 

Thanks a lot :) 

```
  function getValueFromSyntacticEvent(
    event: ChangeEvent<HTMLInputElement>,
  ): string | null {
    return event?.target?.value ?? null
  }

  const handleSearchChange = useCallback(
    flow(
      getValueFromSyntacticEvent,
      debounce((filter: string) => {
        console.warn(filter) // will not be executed!
      }, 100),
    ),
    [],
  )

... component code 

 <TextInput
    id="search"
    label="search"
    placeholder="search something"
    onChange={handleSearchChange}
  />
```

**Comment by Drapaliuk** [comment]
@FocusCookie, hi, did you fix this behavior?

**Reply by FocusCookie** [reply]
No I couldn't find a fix. I never tried it with the latest versions. We moved on and achieved our project.

**Comment by xveganxxxedgex** [comment]
In case this helps anyone, I ran into this but only when mocking the date with `mockdate`. Because the date is set to a fixed time, the timeout was always returning 0 difference in the time for each interval, causing an infinite loop that would never execute the callback function. So I kept a `MockDate.set(new Date(2025, 7, 15, 14, 32, 30));` in the `beforeEach`, then in my test before the timeout would get kicked off, I triggered another `MockDate.set(new Date(2025, 7, 15, 14, 32, 31));` (note the 31 at the end instead of 30) so it would simulate that some time has passed.
